File: samples/C/sum-digits-of-an-integer.c
#include <stdio.h>

int SumDigits(unsigned long long n, const int base) {
    int sum = 0;
    for (; n; n /= base)
    	sum += n % base;
    return sum;
}

int main() {
    printf("%d %d %d %d %d\n",
        SumDigits(1, 10),
        SumDigits(12345, 10),
        SumDigits(123045, 10),
        SumDigits(0xfe, 16),
        SumDigits(0xf0e, 16) );
    return 0;
}

File: samples/Python/sum-digits-of-an-integer.py
def toBaseX(num, base):
    output = []
    while num:
        num, rem = divmod(num, base)
        output.append(rem)
    return output

def sumDigits(num, base=10):
    if base < 2:
        print "Error: Base must be at least 2"
        return
    return sum(toBaseX(num, base))

print sumDigits(1)
print sumDigits(12345)
print sumDigits(123045)
print sumDigits(0xfe, 16)
print sumDigits(0xf0e, 16)
